datagen/procthor_datagen/datagen_runner_train.py

```diff
@@ -0,0 +1,1361 @@
+
+        if any(o["isBroken"] for o in objects_after_first_irs):
+            print("Broken objects after opening")
+            return None, None, None
+
+    valid_pickupables_in_room_after_first_irs, other_pickupables = group_pickupables(
+        env,
+        objects_after_first_irs,
+        object_types_to_not_move,
+        single_room,
+        num_objs_to_move,
+    )
+    if valid_pickupables_in_room_after_first_irs is None:
+        return None, None, None
+
+    # now, choose which objects to move
+
+    random.shuffle(valid_pickupables_in_room_after_first_irs)
+    if num_objs_to_move == 0:
+        unmoved_obj_ids = [
+            o["objectId"] for o in valid_pickupables_in_room_after_first_irs
+        ]
+        moved_obj_ids = []
+    else:
+        unmoved_obj_ids = [
+            o["objectId"]
+            for o in valid_pickupables_in_room_after_first_irs[:-num_objs_to_move]
+        ]
+        moved_obj_ids = [
+            o["objectId"]
+            for o in valid_pickupables_in_room_after_first_irs[-num_objs_to_move:]
+        ]
+
+    if allow_putting_objects_away:
+        # If we're having a really hard time shuffling objects successfully, then let's
+        # move some of the objects we don't care about (i.e. the ones whose position won't change)
+        # into cupboards/drawers/etc so that there is more space.
+
+        hide_kwargs = dict(start_kwargs)
+        hide_kwargs["forceVisible"] = False
+
+        controller.step(
+            "InitialRandomSpawn",
+            **hide_kwargs,
+            excludedObjectIds=moved_obj_ids + excluded_object_ids_for_floor_lamp,
+        )
+        if not controller.last_event.metadata["lastActionSuccess"]:
+            print(controller.last_event.metadata["errorMessage"])
+            return None, None, None
+
+        objects_after_first_irs = copy.deepcopy(env.objects())
+        if any(o["isBroken"] for o in objects_after_first_irs):
+            print("Broken objects after putting objects away")
+            return None, None, None
+
+        (
+            valid_pickupables_in_room_after_first_irs,
+            other_pickupables,
+        ) = group_pickupables(
+            env,
+            objects_after_first_irs,
+            object_types_to_not_move,
+            single_room,
+            num_objs_to_move,
+        )
+        if valid_pickupables_in_room_after_first_irs is None:
+            return None, None, None
+
+        # Check all moved_obj_ids are in valid_pickupables_in_room_after_first_irs
+        valid_ids_in_room = set(
+            o["objectId"] for o in valid_pickupables_in_room_after_first_irs
+        )
+        if any(oid not in valid_ids_in_room for oid in moved_obj_ids):
+            print(
+                "ERROR: Some moved_obj_ids missing from target room after putting objects away"
+            )
+            return None, None, None
+
+    from rearrange.utils import extract_obj_data
```

Comment:
Any reason to not make this a top-level import?

Reply:
I had forgotten it was down there. Good catch.